pkg/kubelet/kubelet_pods.go

```diff
@@ -1785,21 +1795,16 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 			continue
 		}
 
-		if pod.DeletionTimestamp == nil {
-			continue
-		}
```

Comment:
I think I _finally_ got it. This section here was preventing pods that didn't have a DeletionTimestamp set for some reason from getting an unknown termination set.

So, remove this check, and increment restart count by one if this is not set below.